### PR TITLE
[pkg/stanza] SplitFunc naming cleanup

### PR DESCRIPTION
**Pull request by djaglowski**
Another few remaining parts of #26241. This basically follows up on renaming the `multiline` package by renaming a few remaining to "multiline" throughout the codebase.

### CODE REVIEW

File: pkg/stanza/fileconsumer/internal/splitter/custom.go [hunk 1]
```diff
@@ -26,6 +26,6 @@ func NewCustomFactory(splitFunc bufio.SplitFunc, flushPeriod time.Duration) Fact
 }
 
 // Build builds Multiline Splitter struct
-func (f *customFactory) Build() (bufio.SplitFunc, error) {
```
Comment (fatsheep9146) [inline pkg/stanza/fileconsumer/internal/splitter/custom.go:29]: comments should also be updated to SplitFunc

File: pkg/stanza/fileconsumer/internal/splitter/multiline.go [hunk 2]
```diff
@@ -14,35 +14,35 @@ import (
...
+		splitConfig: splitConfig,
+		encoding:    encoding,
+		maxLogSize:  maxLogSize,
+		trimFunc:    trimFunc,
+		flushPeriod: flushPeriod,
 	}
 }
 
 // Build builds Multiline Splitter struct
-func (f *multilineFactory) Build() (bufio.SplitFunc, error) {
```
Comment (fatsheep9146) [inline pkg/stanza/fileconsumer/internal/splitter/multiline.go:44]: same here

File: pkg/stanza/fileconsumer/internal/splitter/multiline_test.go [hunk 3]
```diff
@@ -45,8 +45,8 @@ func TestMultilineBuild(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			factory := NewMultilineFactory(tt.multilineCfg, tt.encoding, tt.maxLogSize, trim.Nop, tt.flushPeriod)
-			got, err := factory.Build()
+			factory := NewSplitFuncFactory(tt.splitConfig, tt.encoding, tt.maxLogSize, trim.Nop, tt.flushPeriod)
+			got, err := factory.SplitFunc()
 			if (err != nil) != tt.wantErr {
 				t.Errorf("Build() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (fatsheep9146) [inline pkg/stanza/fileconsumer/internal/splitter/multiline_test.go:51]: SplitFunc() error

**Comment by djaglowski** [comment]
Thanks for the review @fatsheep9146. I've updated the comments.

**Review by fatsheep9146 — APPROVED** [review]

**Pull request merged**